scripts/view/gas_intelligence.py

```diff
@@ -0,0 +1,200 @@
+                "format": "yyyy-MM-dd hh:mm:ss"
+            },
+            "aggs": {
+                "avgGasHour": {
+                    "avg": {
+                        "field": "gasPrice.num"
+                    }
+                }
+            }
+        }
+    }
+    })
+    return [ x['avgGasHour']['value'] for x in data['aggregations']['hour_bucket']['buckets'] if x['avgGasHour']['value']]
+
+
+# fetch average hourly gas prices over the last specified days
+def fetch_gas_min(network: str, days=1) -> list[float]:
+    es = initialize_elastic(network)
+    data = es.search(index='tx', doc_type='tx', body = {
+        "_source":["timestamp","gasPrice.num"],
+        "query": {
+        "bool": {
+            "must": [{
+                    
+                "range":{
+                    "timestamp": {
+                        "gte": "now-"+ str(days) +"d"
+                    }
+                }}
+                
+            ]
+        }
+        },
+        "aggs": {
+        "minute_bucket": {
+            "date_histogram": {
+                "field": "timestamp",
+                "interval": "1m",
+                "format": "yyyy-MM-dd hh:mm"
+            },
+            "aggs": {
+                "avgGasMin": {
+                    "avg": {
+                        "field": "gasPrice.num"
+                    }
+                }}}
+    }
+    })
+    return [ x['avgGasMin']['value'] for x in data['aggregations']['minute_bucket']['buckets'] if x['avgGasMin']['value']]
+
+
+def is_outlier(points: list[float], thresh=3.5) -> list[bool]:
+    """
+    Returns a boolean array with True if points are outliers and False 
+    otherwise.
+
+    Parameters:
+    -----------
+        points : A numobservations by numdimensions array of observations
+        thresh : The modified z-score to use as a threshold. Observations with
+            a modified z-score (based on the median absolute deviation) greater
+            than this value will be classified as outliers.
+
+    Returns:
+    --------
+        mask : A numobservations-length boolean array.
+
+    References:
+    ----------
+        Boris Iglewicz and David Hoaglin (1993), "Volume 16: How to Detect and
+        Handle Outliers", The ASQC Basic References in Quality Control:
+        Statistical Techniques, Edward F. Mykytka, Ph.D., Editor. 
+    """
+    if len(points.shape) == 1:
+        points = points[:,None]
+    median = np.median(points, axis=0)
+    diff = np.sum((points - median)**2, axis=-1)
+    diff = np.sqrt(diff)
+    med_abs_deviation = np.median(diff)
+
+    modified_z_score = 0.6745 * diff / med_abs_deviation
+    
+    return modified_z_score > thresh
+
+
+# main entry point
+def analyze_gas() -> tuple[float, float, float]:
+    gas_data = []
+
+    # fetch data
+    if HISTORICAL_TIMEFRAME == 'minutes':
+        gas_data = fetch_gas_min(HISTORICAL_URL)
+    elif HISTORICAL_TIMEFRAME == 'hours':
+        gas_data = fetch_gas_hour(HISTORICAL_URL)
```

Comment:
nit: should we return in an `else` branch here for an unhandled timeframe?

Reply:
Yea we should probably default to minutes